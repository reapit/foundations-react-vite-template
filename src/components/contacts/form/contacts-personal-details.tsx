import {
  BodyText,
  InputWrap,
  InputGroup,
  Label,
  Select,
  FormLayout,
  elFadeIn,
  elMb11,
  InputError,
} from '@reapit/elements'
import dayjs from 'dayjs'
import { FC } from 'react'
import { UseFormReturn } from 'react-hook-form'
import { ContactFormSchema } from './contacts-validation-schema'

export const TITLE_OPTIONS = ['Mr', 'Mrs', 'Ms', 'Miss', 'Mx', 'Dr', 'Prof']

interface ContactPersonalProps {
  form: UseFormReturn<ContactFormSchema, any>
}

export const ContactsPersonal: FC<ContactPersonalProps> = ({ form }) => {
  const {
    register,
    formState: { errors },
  } = form

  return (
    <>
      <div className={elMb11}>
        <BodyText hasGreyText hasNoMargin>
          Provide us with some basic personal information about your contact.
        </BodyText>
      </div>
      <FormLayout className={elFadeIn}>
        <InputWrap>
          <InputGroup>
            <Select {...register('title')}>
              <option key="default-option" value="">
                None selected
              </option>
              {TITLE_OPTIONS.map((option) => (
                <option key={option} value={option}>
                  {option}
                </option>
              ))}
            </Select>
            <Label>Please select a title</Label>
            {errors.title?.message && <InputError message={errors.title.message} />}
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="First Name"
            placeholder="Please enter a first name"
            {...register('forename')}
            icon={errors?.forename?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.forename?.message && <InputError message={errors.forename.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Surname"
            placeholder="Please enter a surname"
            {...register('surname')}
            icon={errors?.surname?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.surname?.message && <InputError message={errors.surname.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Date of birth"
            type="date"
            {...register('dateOfBirth')}
            max={dayjs().format('YYYY-MM-DD')}
            icon={errors?.dateOfBirth?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.dateOfBirth?.message && <InputError message={errors.dateOfBirth.message} />}
        </InputWrap>
      </FormLayout>
    </>
  )
}
