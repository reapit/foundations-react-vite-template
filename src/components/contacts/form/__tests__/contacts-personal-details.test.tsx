import { ContactFormSchema } from '../contacts-validation-schema'
import { UseFormReturn } from 'react-hook-form'
import { ContactsPersonal } from '../contacts-personal-details'
import { render } from '../../../../scripts/tests'

const error = {
  message: 'Error',
}

describe('ContactsPersonal', () => {
  it('should match a snapshot', () => {
    const form = {
      register: jest.fn(() => ({})),
      formState: {
        errors: {
          title: error,
          forname: error,
          surname: error,
          dateOfBirth: error,
        },
      },
    } as unknown as UseFormReturn<ContactFormSchema, any>

    expect(render(<ContactsPersonal form={form} />)).toMatchSnapshot()
  })
})
