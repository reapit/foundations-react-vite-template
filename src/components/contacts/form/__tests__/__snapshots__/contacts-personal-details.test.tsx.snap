// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsPersonal should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-93 el-snack-holder"
      />
      <p
        class="el-text-base el-body-text el-has-bold-text el-has-section-margin"
      >
        Provide us with some basic personal information about your contact.
      </p>
      <div
        class="el-fade-in el-form-layout"
      >
        <div
          class="mocked-styled-133 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <select
              aria-label="Select an item from the dropdown list"
              class="mocked-styled-83 el-select"
            >
              <option
                value=""
              >
                None selected
              </option>
              <option
                value="Mr"
              >
                Mr
              </option>
              <option
                value="Mrs"
              >
                Mrs
              </option>
              <option
                value="Ms"
              >
                Ms
              </option>
              <option
                value="Miss"
              >
                Miss
              </option>
              <option
                value="Mx"
              >
                Mx
              </option>
              <option
                value="Dr"
              >
                Dr
              </option>
              <option
                value="Prof"
              >
                Prof
              </option>
            </select>
            <label
              class="mocked-styled-41 el-label"
            >
              Please select a title
            </label>
            <p
              class="mocked-styled-42 el-input-error"
            >
              Error
            </p>
          </div>
        </div>
        <div
          class="mocked-styled-133 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              aria-label="Input type undefined"
              class="mocked-styled-38 el-input"
              id="test-static-id"
              placeholder="Please enter a first name"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              First Name
            </label>
          </div>
        </div>
        <div
          class="mocked-styled-133 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              aria-label="Input type undefined"
              class="el-has-input-error el-input"
              id="test-static-id"
              placeholder="Please enter a surname"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Surname
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-133 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              aria-label="Input type date format dd/mm/yyy max date of 2023-01-01"
              class="el-has-input-error el-input"
              id="test-static-id"
              max="2023-01-01"
              type="date"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Date of birth
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-93 el-snack-holder"
    />
    <p
      class="el-text-base el-body-text el-has-bold-text el-has-section-margin"
    >
      Provide us with some basic personal information about your contact.
    </p>
    <div
      class="el-fade-in el-form-layout"
    >
      <div
        class="mocked-styled-133 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <select
            aria-label="Select an item from the dropdown list"
            class="mocked-styled-83 el-select"
          >
            <option
              value=""
            >
              None selected
            </option>
            <option
              value="Mr"
            >
              Mr
            </option>
            <option
              value="Mrs"
            >
              Mrs
            </option>
            <option
              value="Ms"
            >
              Ms
            </option>
            <option
              value="Miss"
            >
              Miss
            </option>
            <option
              value="Mx"
            >
              Mx
            </option>
            <option
              value="Dr"
            >
              Dr
            </option>
            <option
              value="Prof"
            >
              Prof
            </option>
          </select>
          <label
            class="mocked-styled-41 el-label"
          >
            Please select a title
          </label>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
      </div>
      <div
        class="mocked-styled-133 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            aria-label="Input type undefined"
            class="mocked-styled-38 el-input"
            id="test-static-id"
            placeholder="Please enter a first name"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            First Name
          </label>
        </div>
      </div>
      <div
        class="mocked-styled-133 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            aria-label="Input type undefined"
            class="el-has-input-error el-input"
            id="test-static-id"
            placeholder="Please enter a surname"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Surname
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-133 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            aria-label="Input type date format dd/mm/yyy max date of 2023-01-01"
            class="el-has-input-error el-input"
            id="test-static-id"
            max="2023-01-01"
            type="date"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Date of birth
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
